src/bin/help.ts

```diff
@@ -0,0 +1,172 @@
+import chalk from "chalk";
+
+import { allArgOptions } from "../shared/options/args.js";
+
+interface HelpTextSection {
+	sectionHeading: string;
+	subsections: {
+		flags: Flag[];
+		subheading: string | undefined;
+		warning: string | undefined;
+	}[];
+}
+
+interface Flag {
+	description: string;
+	flag: string;
+	type: string;
+}
+
+interface Option {
+	description: string;
+	docsSection:
+		| "core"
+		| "offline"
+		| "opt-out"
+		| "optional"
+		| "skip-disk"
+		| "skip-net";
+	multiple?: boolean;
+	type: string;
+}
+
+function logHelpTextSection(section: HelpTextSection): void {
+	console.log(" ");
+
+	console.log(`   ${chalk.black.bgGreenBright(section.sectionHeading)}`);
+
+	for (const subsection of section.subsections) {
+		if (subsection.warning) {
+			console.log(chalk.yellow(subsection.warning));
+		}
+
+		if (subsection.subheading) {
+			console.log(chalk.green(subsection.subheading));
+		}
+
+		for (const option of subsection.flags) {
+			const { description, flag, type } = option;
```

Comment:
[Naming] Hmm, the property is called `flags` but the local variable is `option`. That's a little confusing IMO. And it can't be called flag, because the `flag` property is there. We can bypass thinking of a different name by inlining the destructuring:

```suggestion
		for (const { description, flag, type } of subsection.flags) {
```

It does expose a mild naming oddity though - a `Flag` has a `flag`? Maybe, `SectionFlag` for the interface? `SectionOption`? Something like that? _(I'm not confident or picky here)_